### ISSUE TITLE
[BUG] error creating window splits from find-name-dired or fd-dired

### ISSUE THREAD

**Issue by Emantor** [hlissner/doom-emacs#2159]
**Describe the issue**

Sometimes window splits `SPC w v` and `SPC w s` are not working correctly.
This seems to be the case when opening buffers through `find-name-dired`.
The included backtrace is printed when trying to use the shortcuts, however splits with `C-x 3` and `C-x 2` still work.

<details>
<pre>
Debugger entered--Lisp error: (wrong-type-argument window-valid-p #<window 7>)
  window-top-child(#<window 7>)
  window-child(#<window 7>)
  #f(compiled-function (&optional window-or-frame) "Balance the sizes of windows of WINDOW-OR-FRAME.\nWINDOW-OR-FRAME is optional and defaults to the selected frame.\nIf WINDOW-OR-FRAME denotes a frame, balance the sizes of all\nwindows of that frame.  If WINDOW-OR-FRAME denotes a window,\nrecursively balance the sizes of all child windows of that\nwindow." (interactive nil) #<bytecode 0x1aff03>)(#<window 7>)
  apply(#f(compiled-function (&optional window-or-frame) "Balance the sizes of windows of WINDOW-OR-FRAME.\nWINDOW-OR-FRAME is optional and defaults to the selected frame.\nIf WINDOW-OR-FRAME denotes a frame, balance the sizes of all\nwindows of that frame.  If WINDOW-OR-FRAME denotes a window,\nrecursively balance the sizes of all child windows of that\nwindow." (interactive nil) #<bytecode 0x1aff03>) #<window 7>)
  (progn (apply orig-fn args))
  (unwind-protect (progn (apply orig-fn args)) (if popups (progn (let ((origin (selected-window))) (+popup/restore) (if in-popup-p nil (select-window origin))))))
  (let* ((in-popup-p (+popup-buffer-p)) (popups (+popup-windows)) (+popup--inhibit-transient t) +popup--last) (let ((--dolist-tail-- popups)) (while --dolist-tail-- (let ((p (car --dolist-tail--))) (+popup/close p (quote force)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (unwind-protect (progn (apply orig-fn args)) (if popups (progn (let ((origin (selected-window))) (+popup/restore) (if in-popup-p nil (select-window origin)))))))
  +popup-save-a(#f(compiled-function (&optional window-or-frame) "Balance the sizes of windows of WINDOW-OR-FRAME.\nWINDOW-OR-FRAME is optional and defaults to the selected frame.\nIf WINDOW-OR-FRAME denotes a frame, balance the sizes of all\nwindows of that frame.  If WINDOW-OR-FRAME denotes a window,\nrecursively balance the sizes of all child windows of that\nwindow." (interactive nil) #<bytecode 0x1aff03>) #<window 7>)
  apply(+popup-save-a #f(compiled-function (&optional window-or-frame) "Balance the sizes of windows of WINDOW-OR-FRAME.\nWINDOW-OR-FRAME is optional and defaults to the selected frame.\nIf WINDOW-OR-FRAME denotes a frame, balance the sizes of all\nwindows of that frame.  If WINDOW-OR-FRAME denotes a window,\nrecursively balance the sizes of all child windows of that\nwindow." (interactive nil) #<bytecode 0x1aff03>) #<window 7>)
  balance-windows(#<window 7>)
  (progn (balance-windows (window-parent)))
  (if (and (not count) evil-auto-balance-windows) (progn (balance-windows (window-parent))))
  +evil-window-vsplit-a(nil nil)
  apply(+evil-window-vsplit-a (nil nil))
  evil-window-vsplit(nil nil)
  funcall-interactively(evil-window-vsplit nil nil)
  call-interactively(evil-window-vsplit nil nil)
  command-execute(evil-window-vsplit)
</pre>
</detail>

**System information**
<details>

```
emacs   version    26.3
	features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
	build      Aug 29, 2019
	buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
	windowsys  x
	daemonp    server-running
doom    version    2.0.9
	build      HEAD -> develop, origin/develop 318754c4d 2019-12-05 20:48:53 -0500
system  type       gnu/linux
	config     x86_64-pc-linux-gnu
	shell      /bin/zsh
	uname      Linux 5.4.1-arch1-1 #1 SMP PREEMPT Fri, 29 Nov 2019 13:37:24 +0000 x86_64
	path       (~/perl5/bin ~/.local/bin /opt/bin ~/.pyenv/shims ~/work/ptx/git/ptx_dev_tools/ /usr/local/sbin /usr/local/bin /usr/bin /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.3/x86_64-pc-linux-gnu)
config  envfile    nil
	elc-files  0
	modules    (:completion (company +childframe) ivy :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash treemacs (popup +all +defaults) vc-gutter vi-tilde-fringe window-select workspaces ophints :editor multiple-cursors rotate-text (evil +everywhere) file-templates snippets :emacs dired vc :tools eval ansible (lookup +docsets) editorconfig flycheck lsp magit make pass tmux :lang assembly (cc +lsp) data emacs-lisp latex lua markdown (org +attach +babel +capture +export +present) (python +lsp) (rust +lsp) (sh +zsh) web :config (default +bindings +smartparens))
	packages   (n/a)
	elpa       (n/a)
```
</details>

**Comment by hlissner** [comment]
Does this resolve the issue?

```elisp
(after! dired
  (set-popup-rule! "\\^*F\\(?:d\\|ind\\)\\*$" :ignore t))
```

**Comment by Emantor** [comment]
Cursory testing with this added to my config seems to solve the problem, thanks!

**Comment by hlissner** [comment]
I've pushed the fix in 33b7849, so this issue should be resolved. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
